design-documents/add-scope-to-cms-pages.md

```diff
@@ -0,0 +1,94 @@
+
+## [HLD] MC-13907: Add StoreView Scope to CMS Pages
+
+HLD for [MC-13907](https://jira.corp.magento.com/browse/MC-13907): Allow CMS Page to Be Edited and Viewed in Admin Based On StoreView
+
+### Terms
+
+### Overview
+CMS Pages can currently be turned on/off for store views, but the content cannot be customized. Epic MC-13907 will provide support for customizing a Page at the store view level so that users can more easily manage localized content.
+
+**Current schema:**
+
+![](add-scope-to-cms-pages/cms_page_current.png)
+
+The current page design ties a CMS page to one or more store views but this association does not differentiate content by store view, it only defines the availability of the page to a store view.
+
+### Design
+#### Option 1: Introduce a separate content table [Preferred]
+This option would move all page attributes that can be customized to a store view to a separate table, which will include the associated `store_id.` There will be a single "master" page object that retains  `is_active` and store associations at the global level, with all other attributes customizable to the store view level:
+
+![](add-scope-to-cms-pages/cms_page_proposed_option1.png)
+
+##### **Pros:**
+-   Implementation matches that of Dynamic Blocks (banner tables)
+-   Maintains a single `page_id` for each page, regardless of scope
+
+##### **Cons:**
+-   Larger database change will require setup scripts to migrate existing data into the new table structure
+-   Resource model will require more refactoring than Option 2  
+
+##### Scenario: New Page
+When a new page is created, it will default to the All Store Views scope, creating an entry in  `cms_page_content` with `store_id=0`
+
+##### Scenario: Editing a new scope
+After a page is created, changing scope will attempt to load the content matching the current `page_id` and selected  `store_id`. If none exists, the data for  `store_id=0` will be loaded. If changes are made and saved, a new  `cms_page_content` entry will be created to save data under the selected  `store_id`.
+
+##### Scenario: Deleting a page
+Delete a page deletes all content associated with it for all store views.
+
+##### Scenario: Page Builder needs selected store view scope
+Page Builder components (and potentially other extensions) will need to be aware of the selected store view to render store-sensitive data (such as Product List). The store switcher component adds the chosen store_id to the application context and will be available as Page Builder currently expects (via StoreManager) without additional changes.
```

Comment:
Unchecking checkbox changes store view from global to current. Any ideas how to handle this? Would be good to described use cases related to this, I think here can be a lot of corner cases.

Reply:
if you're referring to the Use Default Value checkbox, this just causes the content to be copied from the Default Store View scope. Dynamic block currently functions this way.